__tests__/app.test.js

```diff
@@ -222,11 +221,92 @@ it("Returns 400 if article_id parameter does not reference an existing article i
   return request(app)
   .post('/api/articles/22/comments')
   .send({username : "lurker", body: "Engulfed by a sea of neon lights."})
+  .expect(404)
+  .then(({body : {msg}}) => {
+    expect(msg).toBe("Article not found.");
+  });
+})
+
+it("Returns 400 status code if response URL includes an invalid article_id.", () => {
+  return request(app)
+  .post('/api/articles/id9/comments')
+  .send({username : "lurker", body: "Engulfed by a sea of neon lights."})
   .expect(400)
   .then(({body : {msg}}) => {
-    expect(msg).toBe("Bad request.");
+    expect(msg).toBe("Invalid data type in URL.");
   });
 })
   
 })
-  
\ No newline at end of file
+
+describe("7) PATCH /api/articles/:article_id", () => {
+  it("Responds with 200 status code and the updated article; increases the number of votes on the correct article and by the specified number.", () => {
+    return request(app)
+    .patch('/api/articles/7')
+    .send({ inc_votes: 23 })
+    .expect(200)
+    .then(({body: {updatedArticle}}) => {
+      expect(updatedArticle).toEqual(
+        expect.objectContaining({
+          article_id: 7,
+          votes: 23,
+          author: expect.any(String),
+          title: expect.any(String),
+          topic:expect.any(String),
+          created_at: expect.any(String)
+        })
+      )
+    });
+  })
+  
+  it("Responds with 200 status code and the updated article; decreases the number of votes on the correct article and by the specified number.", () => {
+    return request(app)
+    .patch('/api/articles/1')
+    .send({ inc_votes: -64 })
+    .expect(200)
+    .then(({body: {updatedArticle}}) => {
+      expect(updatedArticle.article_id).toBe(1);
+      expect(updatedArticle.votes).toBe(36);
+    })
+  })
+  
+  it("Responds with 400 status code if request body contains malformed key.", () => {
+    return request(app)
+    .patch('/api/articles/10')
+    .send({ inc_Votes: 4 })
```

Comment:
I'd actually like this test to be more explicitly malformed so `.send({ })` or `.send( { banana: 12 })`